nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -30,22 +30,69 @@ let
   # file provided by services.xserver.displayManager.sessionData.wrapper
   xsessionWrapper = pkgs.writeScript "xsession-wrapper"
     ''
-      #! ${pkgs.bash}/bin/bash
-
-      # Shared environment setup for graphical sessions.
-
-      . /etc/profile
-      if test -f ~/.profile; then
-          source ~/.profile
-      fi
+      #!/bin/sh
+      # Parts of this file are extracted from:
+      #   - kde-workspace (kdm/kfrontend/genkdmconf.c)
+      #     Copyright (C) 2001-2005 Oswald Buddenhagen <ossi@kde.org>
+      #   - sddm (data/scripts/Xsession)
+      #     Copyright (C) 2016 Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
+
+      echo $SHELL > /tmp/shell
```

Comment:
We should not hardcode the path and use mktemp instead.

Reply:
Woops, that was for a local test. Gonna remove it.